asv_bench/benchmarks/algorithms.py

```diff
@@ -89,6 +93,9 @@ def setup_cache(self):
              'timedeltas': pd.timedelta_range('1 day', freq='s', periods=N)})
         df['categories'] = df['strings'].astype('category')
         df.iloc[10:20] = np.nan
+        mi = pd.MultiIndex.from_tuples(list(zip(df['strings'], df['strings'])),
+                                       names=['str', 'int'])
+        df.set_index(mi, inplace=True)
```

Comment:
Would be better if you just did this during construction

Reply:
Removed